src/chart/generateCategoricalChart.js

```diff
@@ -93,23 +94,96 @@ const generateCategoricalChart = ({
       ...defaultProps,
     };
 
+    /**
+     * Calculates how many bars can fit within the diagram space available
+     * @param {object} props - chart props
+     * @param {array} axisComponents - x and y axis components
+     * @return {number} maximum amount of bars that can fit in the diagram
+     */
+    static getMaxBarsForDiagram = (props, axisComponents) => {
+      const { children, width } = props;
+      const barGap = 20;
+      const barSizeList = [10, 15, 20, 25, 30];
```

Comment:
This doesn't seem to apply if alignment is left. Is that expected? If you have one data point and alignment is left, the bar spans almost the entire x axis